src/Models/Services/OatTelemetryUtils.ts

```diff
@@ -15,19 +15,73 @@ export const getNameFromAction = (action: OatPageContextActionType) => {
 export const getOatMetricsForModels = (
     models: DtdlInterface[]
 ): OatImportExportMetrics => {
-    const oatMetrics = getModelMetricsForTelemetry(models);
-    return {
-        modelCount: models.length,
-        relationshipCount: oatMetrics.relationshipCount,
-        inheritanceCount: oatMetrics.inheritanceCount,
-        componentCount: oatMetrics.componentCount,
-        v2ModelCount: oatMetrics.v2ModelCount,
-        v3ModelCount: oatMetrics.v3ModelCount
-    };
+    try {
```

Comment:
in thinking about this again, I think it's better to do this in the useEffects becuase now we will be emitting an event with invalid data. We dont' want to emit an event at all if there is a failure.